core/runtime/binaryen/module/module_factory_impl.cpp

```diff
@@ -32,17 +32,25 @@ namespace kagome::runtime::binaryen {
   }
 
   CompilationOutcome<void> ModuleFactoryImpl::compile(
-      std::filesystem::path path_compiled, BufferView code) const {
+      std::filesystem::path path_compiled,
+      BufferView code,
+      const RuntimeContext::ContextParams &config) const {
+    if (config.wasm_ext_bulk_memory) {
+      return CompilationError{"bulk memory is not supported"};
```

Comment:
Binaryen interpreter doesn't check if bulk memory proposal is allowed,
so this condition will allow wasm with bulk instructions even when `bulk=false`.

Either `ModuleFactory` or `WasmInstrumenter` should check wasm with `wabt::ValidateModule(wabt::ValidateOptions{.bulk = config.bulk})`.

Don't close #2221 or create separate issue for binaryen (and maybe WAVM).

Reply:
What do you suggest if separate issue is created? Remove that check?

Reply:
Add TODO comment with link to that issue.
Remove this check.